packages/react-native-web/src/exports/Image/index.js

```diff
@@ -33,46 +32,28 @@ const STATUS_LOADING = 'LOADING';
 const STATUS_PENDING = 'PENDING';
 const STATUS_IDLE = 'IDLE';
 
-const getImageState = (uri, shouldDisplaySource) => {
-  return shouldDisplaySource ? STATUS_LOADED : uri ? STATUS_PENDING : STATUS_IDLE;
+const getImageState = (source, shouldDisplaySource) => {
+  return shouldDisplaySource ? STATUS_LOADED : source ? STATUS_PENDING : STATUS_IDLE;
 };
 
 const resolveAssetDimensions = source => {
-  if (typeof source === 'number') {
-    const { height, width } = getAssetByID(source);
-    return { height, width };
-  } else if (typeof source === 'object') {
-    const { height, width } = source;
-    return { height, width };
-  }
+  return {
+    height: source.height,
+    width: source.width
+  };
 };
 
-const svgDataUriPattern = /^(data:image\/svg\+xml;utf8,)(.*)/;
-const resolveAssetUri = source => {
-  let uri = '';
-  if (typeof source === 'number') {
-    // get the URI from the packager
-    const asset = getAssetByID(source);
-    const scale = asset.scales[0];
-    const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
-    uri = asset ? `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}` : '';
-  } else if (typeof source === 'string') {
-    uri = source;
-  } else if (source && typeof source.uri === 'string') {
-    uri = source.uri;
+const getCacheUrl = e => {
```

Comment:
This looks a bit fragile. Why do we need to get the url from the native event?

Reply:
If an image is loaded via XHR, the actual image `src` will be replaced by a blob:
https://github.com/necolas/react-native-web/blob/acd3b335518a362842ecfe006a796fcfd2c1a076/packages/react-native-web/src/modules/ImageLoader/index.js#L110

Since the `onload` event is triggered by the image itself, we can retrieve the new blob source in `Image` via the image `src`  in the event handler without passing additional parameters. Though, of course we could  also make this more explicit and pass the event and the new blob url as separate parameters in `onLoad`. What do you prefer?

Reply:
Can we store the blob when we get it, so we don't have to read it back from the native event?

Reply:
Yep, the blob will be stored once the image was loaded:
```ts
ImageUriCache.add(source, getCacheUrl(e));
```

Afterwards it shouldn't be triggered again as long as the source does not change. Or do you mean something else?

Reply:
I'm wondering why we need to get the blob from the event object when we know what it is before that

Reply:
I just reworked that part and now the blob is returned directly from the `ImageLoader`.